tdp/core/collection/collection.py

```diff
@@ -125,9 +117,9 @@ def schema_directory(self) -> Path:
         return self._path / SCHEMA_VARS_DIRECTORY_NAME
 
     @property
-    def dag_nodes(self) -> list[TDPLibDagNodeModel]:
-        """List of DAG files in the YAML format."""
-        return self._dag_nodes  # TODO: should return a generator
+    def dag_nodes(self) -> Generator[TDPLibDagNodeModel, None, None]:
+        """DAG nodes."""
```

Comment:
Should change it to "Read DAG nodes." to highlight that the Generator is created when calling the method.

Reply:
Done.